websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,166 @@
+				presenceData.state = `Strona ${
+					document.location.pathname.split("/")[3]
+				}`;
+			} else {
+				presenceData.details = "Czyta bloga:";
+				presenceData.state = document
+					.querySelectorAll("h1")[0]
+					.textContent.replace("\n", "");
+				presenceData.buttons = [{ label: "Czytaj", url: document.URL }];
+			}
+		} else if (pth === "/gatunki/")
+			presenceData.details = "Przegląda gatunki...";
+		else if (pth.startsWith("/season/")) {
+			presenceData.details = "Przegląda sezon:";
+			presenceData.state = document.querySelectorAll("h1")[0].textContent;
+		} else if (pth === "/sezony/") presenceData.details = "Przegląda sezony...";
+		else if (pth.startsWith("/label/")) {
+			if (pth === "/label/news/") presenceData.details = "Przegląda newsy...";
+			else if (pth === "/label/recenzja/")
+				presenceData.details = "Przegląda recenzje...";
+		} else if (pth === "/ekipa-portalu/")
+			presenceData.details = "Przegląda ekipę portalu...";
+		else if (pth === "/polityka-prywatnosci/")
+			presenceData.details = "Przegląda politykę prywatności...";
+		else if (pth.startsWith("/genres/")) {
+			presenceData.details = "Przegląda gatunek:";
+			presenceData.state = `${document.querySelectorAll("h1")[0].textContent}`;
+			if (pth.includes("page")) {
+				presenceData.state = `${
+					document.querySelectorAll("h1")[0].textContent
+				} (Strona ${document.location.pathname.split("/")[4]})`;
+			}
+		} else if (pth.startsWith("/studio/")) {
+			presenceData.details = "Przegląda studio:";
+			presenceData.state = `${document.querySelectorAll("h1")[0].textContent}`;
+			if (pth.includes("page")) {
+				presenceData.state = `${
+					document.querySelectorAll("h1")[0].textContent
+				} (Strona ${document.location.pathname.split("/")[4]})`;
+			}
+		} else if (
+			document.querySelectorAll("span.year")[0].textContent &&
+			document
+				.querySelectorAll("span.year")[0]
+				.textContent.includes("Dodane przez")
+		) {
+			presenceData.details = `Ogląda: ${
+				document.querySelector(
+					"body > div > div > div > div > div > div > div > h3"
+				).textContent
+			}`;
+			presenceData.state = `Odcinek:${
+				document
+					.querySelector(
+						"body > div > div > div > div > div > div > div > span"
+					)
+					.textContent.split("-")[1]
+					.split("/")[0]
+			}`;
+			presenceData.buttons = [
+				{ label: "Oglądaj", url: document.URL },
+				{
+					label: "Cała seria",
+					url: document
+						.querySelector(
+							"body > div > div > div > div > div > div > div > h3 > a"
+						)
+						.getAttribute("href"),
+				},
+			];
+			presenceData.largeImageKey = document
+				.querySelector("img.ts-post-image")
+				.getAttribute("src");
+			[presenceData.startTimestamp, presenceData.endTimestamp] =
+				presence.getTimestamps(video.current, video.duration);
+
+			presenceData.smallImageKey = "play";
+			presenceData.smallImageText = "Odtwarzanie";
+			if (video.paused) {
+				presenceData.smallImageKey = "pause";
```

Comment:
Replace `smallImageKey` with an Imgur link, which is preferred over Discord asset names for easier future maintainability.